### PR TITLE
Fixed some typographical errors, grammatical errors, verbose sentences, and inconsistencies in the 'start' section of the doc 

### PR DESCRIPTION
**Pull request by dibia27**
## Explanation

This PR is needed to fix typographical errors, grammatical errors, and punctuation errors. I also reconstructed some verbose sentences to make comprehension easier for users of the document, and I added the "Overview" H2 heading to some sections of the document to break up the document into sections for easier navigation of the document. I made all these changes taking into consideration the best practices for technical writing and also making the document more of a tool for users.

## Related issue

typographical errors, grammatical errors, verbose sentences, and inconsistencies.

## What type of PR is this

> /kind bug

## Proposed Changes

The changes made in the documentation follow technical writing best practices and these changes will guarantee professionalism in the documentation of WasmEdge. The proposed changes will make the documentation easy for users to comprehend, and it will serve as a user-friendly guide thereby turning the document into a  self-help tool.

### CODE REVIEW

**Comment by alabulei1** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: docs/start/build-and-run/aot.md [hunk 1]
```diff
@@ -121,7 +121,7 @@ This AOT-compiled WASM file is compatible with any WebAssembly runtime. However,
 
 <!-- prettier-ignore -->
 :::note
-On MacOS platforms, the universal WASM format will `bus error` in execution. By default, the `wasmedge compile` tool optimizes the WASM in the `O2` level. We are trying to fix this issue. For working around, please use the shared library output format instead.
+On MacOS platforms, the universal WASM format will `bus error` in execution. By default, the `wasmedge compile` tool optimizes the WASM in the `O2` level. We are trying to fix this issue. For working around, use the shared library output format instead.
```
Comment (adithyaakrishna) [inline docs/start/build-and-run/aot.md:124]:
```suggestion
On MacOS platforms, the universal WASM format will `bus error` in execution. By default, the `wasmedge compile` tool optimizes the WASM in the `O2` level. We are trying to fix this issue. For working around, please use the shared library output format instead.
```

File: docs/start/build-and-run/aot.md [hunk 2]
```diff
@@ -131,7 +131,7 @@ wasmedge app_aot.wasm
 
 ## Output Format: Shared Library
 
-Users can assign the shared library extension for the output files (`.so` on Linux, `.dylib` on MacOS, and `.dll` on Windows) to generate the shared library output format output.
+You can assign the shared library extension for the output files (`.so` on Linux, `.dylib` on MacOS, and `.dll` on Windows) to generate the shared library output format output.
```
Comment (adithyaakrishna) [inline docs/start/build-and-run/aot.md:134]:
```suggestion
Users can assign the shared library extension for the output files (`.so` on Linux, `.dylib` on MacOS, and `.dll` on Windows) to generate the shared library output format output.
```

**Comment by adithyaakrishna** [comment]
@dibia27 Some changes are not needed, for example the ones suggested above. Could you please revert the unnecessary changes?

File: docs/start/getting-started/quick_start_redhat.md [hunk 3]
```diff
@@ -86,14 +92,14 @@ $ podman --runtime /usr/bin/crun-wasm run -dp 8080:8080 --platform=wasi/wasm -t
...
-From another terminal window, do the following.
+From another terminal window, do the following:
 
 ```bash
 $ curl http://localhost:8080/echo -X POST -d "Hello WasmEdge"
 Hello WasmEdge
 ```
 
-To learn more about how to run JavaScript apps in WasmEdge.
+To learn more about how to run JavaScript apps in WasmEdge, see the following:
```
Comment (adithyaakrishna) [inline docs/start/getting-started/quick_start_redhat.md:102]:
```suggestion
To learn more about how to run JavaScript apps in WasmEdge,
```

File: docs/start/install.md [hunk 4]
```diff
@@ -4,7 +4,9 @@ sidebar_position: 2
 
 # Install and uninstall WasmEdge
 
-This chapter will discuss ways to install and uninstall the WasmEdge Runtime on various OSes and platforms. We will cover how to install plug-ins to WasmEdge.
+## Overview
+
+In this chapter, you will learn various ways to install and uninstall the WasmEdge Runtime on various OSes and platforms, and you will learn how to install plug-ins to WasmEdge.
```
Comment (adithyaakrishna) [inline docs/start/install.md:9]: Revert these changes

File: docs/start/install.md [hunk 5]
```diff
@@ -17,46 +19,46 @@ You can install the WasmEdge Runtime on any generic Linux and MacOS platforms. I
 
 ### Generic Linux and MacOS
 
-The easiest way to install WasmEdge is to run the following command. Your system should have `git` and `curl` as prerequisites.
+Your system should have `git` and `curl` as prerequisites. The easiest way to install WasmEdge is to run the following command:
```
Comment (adithyaakrishna) [inline docs/start/install.md:22]: Could you please revert these changes as well

File: docs/start/install.md [hunk 6]
```diff
@@ -17,46 +19,46 @@ You can install the WasmEdge Runtime on any generic Linux and MacOS platforms. I
...
+WasmEdge is installed in the `$HOME/.wasmedge` directory by default. You can install it into a system directory such as `/usr/local` to make it available to all users. To specify an install directory, run the `install.sh` script with the `-p` flag. You will need to run the following commands as the `root` user or `sudo` since they are written into the system directories:
 
 ```bash
 curl -sSf https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- -p /usr/local
 ```
 
 #### Install a specific version of WasmEdge
 
-The WasmEdge installer script will install the latest official release by default. You could install a specific version of WasmEdge, including pre-releases or old releases by passing the `-v` argument to the installer script. Here is an example.
+The WasmEdge installer script will install the latest official release by default. You could install a specific version of WasmEdge, including pre-releases or old releases by passing the `-v` argument to the installer script. Here is an example:
```
Comment (adithyaakrishna) [inline docs/start/install.md:44]:
```suggestion
The WasmEdge installer script will install the latest official release by default. You could install a specific version of WasmEdge, including pre-releases or old releases by passing the `-v` argument to the installer script. Here is an example,
```

File: docs/start/install.md [hunk 7]
```diff
@@ -17,46 +19,46 @@ You can install the WasmEdge Runtime on any generic Linux and MacOS platforms. I
...
 
 ```bash
 VERSION={{ wasmedge_version }}
 curl -sSf https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- -v $VERSION
 ```
 
-Suppose you are interested in the latest builds from the `HEAD` of the `master` branch, which is basically WasmEdge's nightly builds. In that case, you can download the release package directly from our Github Action's CI artifact. [Here is an example](https://github.com/WasmEdge/WasmEdge/actions/runs/2969775464#artifacts).
+If you are interested in the latest builds from the `HEAD` of the `master` branch, which is basically WasmEdge's nightly builds,you can download the release package directly from our Github Action's CI artifact. [Here is an example](https://github.com/WasmEdge/WasmEdge/actions/runs/2969775464#artifacts).
```
Comment (adithyaakrishna) [inline docs/start/install.md:51]: Could you please revert these changes as well

File: docs/start/install.md [hunk 8]
```diff
@@ -17,46 +19,46 @@ You can install the WasmEdge Runtime on any generic Linux and MacOS platforms. I
...
 
 #### Install WasmEdge with plug-ins
 
-WasmEdge plug-ins are pre-built native modules that provide additional functionalities to the WasmEdge Runtime. To install plug-ins with the runtime, you can pass the `--plugins` parameter in the installer. For example, the command below installs the `WASI-NN TensorFlow-Lite backend` plug-in, which allows WasmEdge apps to run inference on Tensorflow-Lite models with the `WASI-NN` proposal.
+WasmEdge plug-ins are pre-built native modules that provide additional functionalities to the WasmEdge Runtime. To install plug-ins with the runtime, you can pass the `--plugins` parameter in the installer. For example, the command below installs the `WASI-NN TensorFlow-Lite backend` plug-in, which allows WasmEdge apps to run inference on Tensorflow-Lite models with the `WASI-NN` proposal:
```
Comment (adithyaakrishna) [inline docs/start/install.md:55]:
```suggestion
WasmEdge plug-ins are pre-built native modules that provide additional functionalities to the WasmEdge Runtime. To install plug-ins with the runtime, you can pass the `--plugins` parameter in the installer. For example, the command below installs the `WASI-NN TensorFlow-Lite backend` plug-in, which allows WasmEdge apps to run inference on Tensorflow-Lite models with the `WASI-NN` proposal,
```

File: docs/start/build-and-run/aot.md (hunk 1 above)
Comment (dibia27) [reply docs/start/build-and-run/aot.md:124]: Hi @adithyaakrishna, for this suggested change of adding back the word 'please,' I honestly think it's not a necessary change. The reason I made the change to remove the word 'please' from the sentence is because in technical writing, the word 'please' is considered redundant. Docs serve as strict instructions for users, so the place of 'please' in docs is questionable. I'll kindly request you do some research on using the word 'please' in technical writing and Docs generally, and we could have a final conclusion on this. If you also insist on adding 'please,' I'd love to know specifically your decision for using the word 'please' in the sentence. Thanks

File: docs/start/install.md (hunk 4 above)
Comment (dibia27) [reply docs/start/install.md:9]: Noted.

File: docs/start/install.md (hunk 5 above)
Comment (dibia27) [reply docs/start/install.md:19]: Noted.

File: docs/start/install.md (hunk 7 above)
Comment (dibia27) [reply docs/start/install.md:49]: Noted

**Comment by dibia27** [comment]
@adithyaakrishna I've resolved most of the suggestions you made, and I'll be looking forward to your response on the remaining change.

**Comment by alabulei1** [comment]
Hi @dibia27 

Overall, I couldn't much difference before and after your PR. Could you please keep fixing the typo changes?

**Comment by dibia27** [comment]
Hi @alabulei1, while I'd love to continue working on this project, I currently have a lot of tasks I'm attending to right now, and I would not be available to continue this. Thank you.
